File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/transtale/Translator.java
package tw.nekomimi.nekogram.transtale;
import java.util.Locale;
import org.jetbrains.annotations.NotNull;
import tw.nekomimi.nekogram.cc.CCTarget;
public class Translator {
    public static void translate(String text, TranslateCallBack callback) {}
    public static void translate(String text, String to, TranslateCallBack callback) {}
    public static void translate(String text, String to, String from, TranslateCallBack callback) {}
    public static void translate(String text, Locale to, TranslateCallBack callback) {}
    public static void translate(String text, Locale to, Locale from, TranslateCallBack callback) {}
    // Overloads accepting Companion.TranslateCallBack
    public static void translate(String text, Companion.TranslateCallBack callback) {}
    public static void translate(String text, String to, Companion.TranslateCallBack callback) {}
    public static void translate(String text, String to, String from, Companion.TranslateCallBack callback) {}
    public static void translate(String text, Locale to, Companion.TranslateCallBack callback) {}
    public static void translate(String text, Locale to, Locale from, Companion.TranslateCallBack callback) {}
    public static void translate(Locale to, String text, Companion.TranslateCallBack callback) {}
    public static void translate(Locale to, Locale from, Companion.TranslateCallBack callback) {}
    public static String getMessagePlainText(Object msg) { return ""; }
    public static Locale forLocale(Locale locale) { return locale; }
    public static Locale getChatLanguage(long chatId, Locale defaultLocale) { return defaultLocale; }
    public static void saveChatLanguage(long chatId, Locale locale) {}
    public static CCTarget getChatCCTarget(long chatId, String lang) { return null; }
    public static void showTargetLangSelect(Object menuItem, boolean b, Object callback) {}
    public static void showCCTargetSelect(Object menuItem, Object callback) {}
    public static void showTransFailedDialog(Object context, boolean unsupported, String message, Runnable onRetry) {}
    public static int currentAppId() { return 0; }
    public static class Companion {
        public interface TranslateCallBack {
            void onSuccess(@NotNull String translation);
            void onFailed(boolean unsupported, @NotNull String message);
        }
    }
    public interface TranslateCallBack {
        void onSuccess(@NotNull String translation);
        void onFailed(boolean unsupported, @NotNull String message);
    }
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/transtale/Translator.kt
package tw.nekomimi.nekogram.transtale

import android.view.View
import org.telegram.messenger.LocaleController
import java.util.*

// Extension properties kept for compatibility
val String.code2Locale: Locale
    get() {
        if (this.isBlank()) return LocaleController.getInstance().currentLocale
        val args = replace('-', '_').split('_')
        return if (args.size == 1) Locale(args[0]) else Locale(args[0], args[1])
    }

val Locale.locale2code: String
    get() = if (country.isNullOrBlank()) language else "$language-$country"

interface Translator {
    suspend fun doTranslate(from: String, to: String, query: String): String

    companion object {
        const val providerGoogle = 1
        const val providerGoogleCN = 2
        const val providerYandex = 3
        const val providerLingo = 4
        const val providerMicrosoft = 5
        const val providerYouDao = 6
        const val providerDeepL = 7
        const val providerTelegram = 8
        const val providerTranSmart = 9

        @JvmStatic
        @JvmOverloads
        fun translate(to: Locale = LocaleController.getInstance().currentLocale, query: String, translateCallBack: TranslateCallBack) {
            // Translation disabled in Tony Chat
            translateCallBack.onFailed(true, "Translation not available")
        }

        @JvmStatic
        @JvmOverloads
        fun showTargetLangSelect(anchor: View, input: Boolean = false, full: Boolean = false, callback: (Locale) -> Unit = {}) {
            // No-op
        }

        // Overload for 3 argument calls from Java
        @JvmStatic
        fun showTargetLangSelect(anchor: View, input: Boolean, callback: (Locale) -> Unit) {
            // No-op
        }

        @JvmStatic
        @JvmOverloads
        fun showCCTargetSelect(anchor: View, input: Boolean = true, callback: (String) -> Unit = {}) {
            // No-op
        }

        // Overload for 2 argument calls from Java
        @JvmStatic
        fun showCCTargetSelect(anchor: View, callback: (String) -> Unit) {
            // No-op
        }

        interface TranslateCallBack {
            fun onSuccess(translation: String)
            fun onFailed(unsupported: Boolean, message: String)
        }
    }
}
